### DISCUSSION TITLE
June 4th 2022 - Know Devices and Visitor Profiles are displaying

### DISCUSSION THREAD

**Discussion by jonathonbyrdziak** [merchantprotocol/theadslab#63] [Announcements]
### The Problem
The known devices and visitor profiles pages were breaking with too much data. It was also taking an exceptionally long time to load these pages.

### The Solution
- I rolled out an update today which fixes all of the errors with these two pages. You shouldn't see any code errors, but you still might see a timeout if you are trying to sort and reorder the data. 
- I limited the pagination to only display the first page, with a default of 10 records. I will be enhancing this to work with the javascript datatables to prevent the mass load on the client browser
- Implemented some database indexes on key columns to speed up the returned data.

**Comment by abadger13** [comment]
@jonathonbyrdziak HUGE thank you. This is amazing. Fantastic. Thank you.
